lib/ui/painting/picture.cc

```diff
@@ -79,66 +81,126 @@ static sk_sp<DlImage> CreateDeferredImage(
       raster_task_runner, std::move(unref_queue));
 }
 
-// static
-void Picture::RasterizeToImageSync(sk_sp<DisplayList> display_list,
-                                   uint32_t width,
+void Picture::RasterizeToImageSync(uint32_t width,
                                    uint32_t height,
                                    Dart_Handle raw_image_handle) {
   auto* dart_state = UIDartState::Current();
   if (!dart_state) {
     return;
   }
+  auto raster_task_runner = dart_state->GetTaskRunners().GetRasterTaskRunner();
   auto unref_queue = dart_state->GetSkiaUnrefQueue();
   auto snapshot_delegate = dart_state->GetSnapshotDelegate();
-  auto raster_task_runner = dart_state->GetTaskRunners().GetRasterTaskRunner();
 
   auto image = CanvasImage::Create();
   auto dl_image = CreateDeferredImage(
-      dart_state->IsImpellerEnabled(), std::move(display_list), width, height,
-      std::move(snapshot_delegate), std::move(raster_task_runner),
+      dart_state->IsImpellerEnabled(), display_list(), impeller_picture(),
+      width, height, std::move(snapshot_delegate), raster_task_runner,
       std::move(unref_queue));
+
   image->set_image(dl_image);
   image->AssociateWithDartWrapper(raw_image_handle);
 }
 
 void Picture::dispose() {
-  display_list_.reset();
+  picture_ = sk_sp<DisplayList>(nullptr);
   ClearDartWrapper();
 }
 
 size_t Picture::GetAllocationSize() const {
-  if (display_list_) {
-    return display_list_->bytes() + sizeof(Picture);
-  } else {
-    return sizeof(Picture);
+  auto size = sizeof(Picture);
+  if (display_list()) {
+    size += display_list()->bytes();
   }
+  // TODO(dnfield): Add support to EntityPass to get its allocation size.
```

Comment:
What are we using the allocation size for? Is this necesary to support?

Reply:
It's exposed in https://api.flutter.dev/flutter/dart-ui/Picture/approximateBytesUsed.html

Reply:
:(

entity count * 100?